cdap-api/src/main/java/co/cask/cdap/api/lineage/Read.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents a read operation from a data source into a collection of fields.
+ */
+public class Read extends AbstractOperation {
+  private final String source;
+  private final Map<String, String> sourceProperties;
+  private final List<String> fields;
+
+  /**
+   * Create an instance of READ operation
+   * @param name the name of the operation
+   * @param description the description associated with the operation
+   * @param source the source for the operation
+   * @param fields the array of fields generated
+   */
+  public Read(String name, String description, String source, String ... fields) {
+    this(name, description, source, new HashMap<>(), fields);
+  }
+
+  /**
+   * Create an instance of READ operation
+   * @param name the name of the operation
+   * @param description the description associated with the operation
```

Comment:
So is description required? I feel that in the previous version of this API (the builder), it was optional. Should it be @Nullable then?